ocaml-cohttp-eio/lib/scenarios.ml

```diff
@@ -0,0 +1,207 @@
+open Eio.Std
+open Cohttp_eio
+open Utils
+
+let make_request ~port ~path ?query ~sw ~net () =
+  let uri =
+    Uri.with_uri ~scheme:(Some "http") ~host:(Some "localhost")
+      ~port:(Some port) ~path:(Some path) ~query Uri.empty
+  in
+  let client = Client.make ~https:None net in
+  try
+    let resp, body = Client.get client ~sw uri in
+    if Http.Status.compare resp.status `OK = 0 then
+      let content = Eio.Buf_read.(parse_exn take_all) body ~max_size:max_int in
+      (resp.status, Some content)
+    else (resp.status, None)
+  with _exn -> Fiber.await_cancel ()
```

Comment:
A problem with this wait-on-error pattern is that if all requests fail then the program will just hang instead of reporting an error.

An alternative would be to raise a `Success` exception on success and start the fibers with `both` or `all` instead of `any`, e.g something like:

```ocaml
let wrap_success f () =
  match f () with
  | x -> raise (Success x)
  | exception (Eio.Io _ as e) -> traceln "%a" Eio.Exn.pp e

let any_non_failing fns =
  try
    Fiber.all (List.map wrap_success fns);
    failwith "All failed!"
  with Success x -> x
```

Reply:
Ah, that's a smart way of handling this. Thanks!